### PR TITLE
Dashboard Revamp

### PR DESCRIPTION
**Pull request by MVarshini**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR revamps the current Dashboard 

-  Updated the Layout
-  Added Table component with Sorting, Pagination
-  Integrated with CPT_JOBS API
-  Updated the Doc file Readme.md 

## Related Tickets & Documents

- Related Issue # [PANDA-361](https://issues.redhat.com/browse/PANDA-361)
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please describe the System Under Test.
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by vishnuchalla** [comment]
@MVarshini Can you update the `local-compose.sh` file so that we can verify these changes? Looks like the existing one is not compatible.

File: frontend/src/utils/apiConstants.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const getUrl = () => {
+  const { hostname, protocol } = window.location;
+  return hostname === "localhost"
+    ? "http://localhost:8000"
+    : `${protocol}//${hostname}`;
+};
+
+export const BASE_URL = getUrl();
+
+export const OCP_JOBS_API_V1 = "/api/v1/ocp/jobs";
```
Comment (vishnuchalla) [inline frontend/src/utils/apiConstants.js:10]: I am not sure where these URLs are being used in the code.

File: frontend/src/actions/homeActions.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import * as API_ROUTES from "@/utils/apiConstants";
+import * as TYPES from "@/actions/types.js";
+
+import API from "@/utils/axiosInstance";
+import { DEFAULT_PER_PAGE } from "@/assets/constants/paginationConstants";
+import { showFailureToast } from "@/actions/toastActions";
+
+export const fetchOCPJobsData = () => async (dispatch, getState) => {
```
Comment (vishnuchalla) [inline frontend/src/actions/homeActions.js:8]: This is the only stub making calls to the backend. But I am not sure if it is able to plugin OCP and Quay data as well to the frontend. If not here can you please help me understand where is it happening in the frontend code?

File: frontend/src/utils/apiConstants.js (hunk 1 above)
Comment (MVarshini) [reply frontend/src/utils/apiConstants.js:10]: As discussed, these URLs will be used in the future.

File: frontend/src/actions/homeActions.js (hunk 2 above)
Comment (MVarshini) [reply frontend/src/actions/homeActions.js:8]: This PR is for Home page so for OCP and Quay data new PRs will be raised.

**Comment by MVarshini** [comment]
> @MVarshini Can you please update the `local-compose.sh` file so that we can verify these changes? Looks like the existing one is not compatible.

@vishnuchalla Updated the file.

**Comment by vishnuchalla** [comment]
> > @MVarshini Can you please update the `local-compose.sh` file so that we can verify these changes? Looks like the existing one is not compatible.
> 
> @vishnuchalla Updated the file.

I still come across this error when I run `sh local-compose.sh`
```
[1/4] Resolving packages...
[2/4] Fetching packages...
error vite@5.2.9: The engine "node" is incompatible with this module. Expected version "^18.0.0 || >=20.0.0". Got "16.20.2"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: building at STEP "RUN yarn install": while running runtime: exit status 1
87f2197c97e62613568d0ee74158e85d4318c95da5152a2275b5244e79e1b474
Trying to pull localhost/frontend:latest...
WARN[0000] Failed, retrying in 1s ... (1/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
WARN[0001] Failed, retrying in 1s ... (2/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
WARN[0002] Failed, retrying in 1s ... (3/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused
```

**Comment by vishnuchalla** [comment]
> > > @MVarshini Can you please update the `local-compose.sh` file so that we can verify these changes? Looks like the existing one is not compatible.
> > 
> > 
> > @vishnuchalla Updated the file.
> 
> I still come across this error when I run `sh local-compose.sh`
> 
> ```
> [1/4] Resolving packages...
> [2/4] Fetching packages...
> error vite@5.2.9: The engine "node" is incompatible with this module. Expected version "^18.0.0 || >=20.0.0". Got "16.20.2"
> error Found incompatible module.
> info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
> Error: building at STEP "RUN yarn install": while running runtime: exit status 1
> 87f2197c97e62613568d0ee74158e85d4318c95da5152a2275b5244e79e1b474
> Trying to pull localhost/frontend:latest...
> WARN[0000] Failed, retrying in 1s ... (1/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
> WARN[0001] Failed, retrying in 1s ... (2/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
> WARN[0002] Failed, retrying in 1s ... (3/3). Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused 
> Error: initializing source docker://localhost/frontend:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp [::1]:443: connect: connection refused
> ```

Checked again. Works fine.

**Comment by jtaleric** [comment]
What is our status on this integration?

**Comment by vishnuchalla** [comment]
@chentex and I had to chat to get these in after the telco changes. @MVarshini would you mind updating this PR please?

**Review by jtaleric — CHANGES_REQUESTED** [review]
@MVarshini I am unable to test this change, `local-compose` seems to run, however the frontend pod doesn't seem to be serving out anything? 

```

> frontend@0.0.0 dev
> vite


  VITE v5.2.7  ready in 156 ms

  ➜  Local:   http://localhost:3000/
  ➜  Network: use --host to expose
```

Is the containerfile for the frontend up to date?

**Comment by MVarshini** [comment]
> @MVarshini I am unable to test this change, `local-compose` seems to run, however the frontend pod doesn't seem to be serving out anything?
> 
> ```
> 
> > frontend@0.0.0 dev
> > vite
> 
> 
>   VITE v5.2.7  ready in 156 ms
> 
>   ➜  Local:   http://localhost:3000/
>   ➜  Network: use --host to expose
> ```
> 
> Is the containerfile for the frontend up to date?

@jtaleric The container file is up to date. Try to hit http://localhost:3000/ on the browser if it doesn't work let's it offline and check.

**Comment by jtaleric** [comment]
@MVarshini Thanks for the fix to the local-compose -- in the past we never needed hostNetwork.

There are still aspects of the new dashboard that seem broken for me. 
image
When I select OCP, the list of tests are not updating to just OCP workloads.

image
When I click the menu button, I see a "hello" message. 

I don't want to rush things, but is this PR ready?

**Comment by MVarshini** [comment]
> @MVarshini Thanks for the fix to the local-compose -- in the past we never needed hostNetwork.
> 
> There are still aspects of the new dashboard that seem broken for me. image When I select OCP, the list of tests are not updating to just OCP workloads.
> 
> image When I click the menu button, I see a "hello" message.
> 
> I don't want to rush things, but is this PR ready?

It has only the Home Page Table and OCP page hasn't been included here.

Also, Side menu toggle is a bug and it has been fixed in this PR [#102](https://github.com/cloud-bulldozer/cpt-dashboard/pull/102) which has both Home and OCP pages with Filter and Summary Tab

File: frontend/src/components/organisms/Header/index.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { showToast } from "@/actions/toastActions";
+import { toggleSideMenu } from "@/actions/sideMenuActions";
+
+const Header = () => {
+  const isSideMenuOpen = useSelector((state) => state.sidemenu.isSidebarOpen);
+
+  const dispatch = useDispatch();
+  const onSidebarToggle = () => {
+    dispatch(toggleSideMenu(false));
+    dispatch(showToast("success", "Hello"));
```
Comment (vishnuchalla) [inline frontend/src/components/organisms/Header/index.jsx:25]: nit: Please remove this message.

**Review by vishnuchalla — APPROVED** [review]
Considering as this revamp has to go in multiple PRs, I am okay with merging this PR into separate branch as an initial step and having rest of the parts integrated in the upcoming ones. For example this PR only has home page and the layout set but doesn't actually filter out stuff and show product specific data (.i.e OCP, Quay etc) which is fine. Hopefully all these items will get addressed in https://github.com/cloud-bulldozer/cpt-dashboard/pull/102. 
cc: @chentex

File: frontend/src/components/organisms/Header/index.jsx (hunk 3 above)
Comment (MVarshini) [reply frontend/src/components/organisms/Header/index.jsx:25]: Addressed in PR[102](https://github.com/cloud-bulldozer/cpt-dashboard/pull/102)

**Review by chentex — APPROVED** [review]
Thank you @MVarshini, merging this one to the revamp branch and test the #102

**Pull request merged**